flink-runtime/src/test/java/org/apache/flink/runtime/util/config/memory/ManagedMemoryUtilsTest.java

```diff
@@ -287,22 +297,27 @@ public void testUseCaseWeightsConfiguredWithConsistentValue() {
         ManagedMemoryUtils.validateUseCaseWeightsNotConflict(existingWeights, newWeights);
     }
 
-    @Test(expected = IllegalStateException.class)
-    public void testUseCaseWeightsConfiguredWithConflictValue() {
-        final Map<ManagedMemoryUseCase, Integer> existingWeights =
-                new HashMap<ManagedMemoryUseCase, Integer>() {
-                    {
-                        put(ManagedMemoryUseCase.OPERATOR, 123);
-                    }
-                };
+    @Test
+    void testUseCaseWeightsConfiguredWithConflictValue() {
+        assertThatThrownBy(
+                        () -> {
+                            final Map<ManagedMemoryUseCase, Integer> existingWeights =
+                                    new HashMap<ManagedMemoryUseCase, Integer>() {
+                                        {
+                                            put(ManagedMemoryUseCase.OPERATOR, 123);
+                                        }
+                                    };
 
-        final Map<ManagedMemoryUseCase, Integer> newWeights =
-                new HashMap<ManagedMemoryUseCase, Integer>() {
-                    {
-                        put(ManagedMemoryUseCase.OPERATOR, 456);
-                    }
-                };
+                            final Map<ManagedMemoryUseCase, Integer> newWeights =
+                                    new HashMap<ManagedMemoryUseCase, Integer>() {
+                                        {
+                                            put(ManagedMemoryUseCase.OPERATOR, 456);
+                                        }
+                                    };
 
-        ManagedMemoryUtils.validateUseCaseWeightsNotConflict(existingWeights, newWeights);
+                            ManagedMemoryUtils.validateUseCaseWeightsNotConflict(
+                                    existingWeights, newWeights);
+                        })
+                .isInstanceOf(IllegalStateException.class);
```

Comment:
Same to the last comment, the `existingWeights` and `newWeights` should be removed from this block.